src/basicio.cpp

```diff
@@ -1931,32 +1883,14 @@ namespace Exiv2 {
             case BasicIo::end: newIdx = p_->size_ + offset; break;
         }
 
+        /// \todo in previous ::seek() overload there was the following comment:
+        ///  #1198.  Don't return 1 when asked to seek past EOF.  Stay calm and set eof_
+        /// And the following line was commented. Check out this part of the code
```

Comment:
This is the only change that I think it might be a bit risky. The previous 2 `RemoteIo::seek()` overloads had slightly different code.

Any ideas about how we could test this? I think somebody added some python tests to exercise the RemoteIo code, but I did not pay too much attention there.

Reply:
@piponazo  I thought @kevinbackhouse removed `isMalloced_` from the code base last week.

Reply:
I removed a different `isMalloced_`:

https://github.com/Exiv2/exiv2/blob/c28e501b91f55202dd6c99081b022582a399657a/src/tiffcomposite_int.hpp#L549

I think the `isMalloced_` in `basicio.cpp` looks ok because I don't see any copy constructors or operators. In `TiffEntryBase` I was worried that the ownership of the buffer was unclear.

Reply:
I think it would be safer to make sure that `0 <= idx_` and `idx_ <= size_` are always true. Because code like this doesn't look like it was designed to handle the case where `idx_` is out-of-bounds:

https://github.com/Exiv2/exiv2/blob/c28e501b91f55202dd6c99081b022582a399657a/src/basicio.cpp#L1858-L1866

Reply:
I took your suggestion and added the check for `0 <= idx_` and also set the EOF variable to true in case the newIdx is bigger than the size.

Reply:
> Any ideas about how we could test this? I think somebody added some python tests to exercise the RemoteIo code, but I did not pay too much attention there.

@piponazo There was some python to test RemoteIo.  I think it was focused on getting complete files and the like and could exercise different http verbs (the request methods: get/head/etc).  We've deprecated the WebReady ssh code.  I don't know the status of ftp.  Nobody has ever reported any issue about ssh or ftp, so I don't think it's being used.

http is tested in the bash tests by powering up the python web-server.  There's code in tests/bash_tests/utils.py to start/stop the python web server.

Can you write a C++ unit test (in bin/unit_tests) to test `BasicIo::seek()` on local storage?  When that's working, we can discuss how best to power up the python web-server and perform the test on a remote file.

Reply:
Hi Robin. We already have such unit tests for MemIo and FileIo. You can see those tests in the following files:

```
exiv2/unitTests/test_basicio.cpp
exiv2/unitTests/test_FileIo.cpp
```

I have actually revisited some of those tests, made some improvements there (specially in terms of test naming) and add few new tests.

Reply:
Careful API analysis and code elimination in classes FileIo and MemIo is of great value.  Probably 99.9% of BasicIo use occurs in `classes FileIo` and `class MemIo`.

How about opening an issue: "Unit Test RemoteIo".  Just _**punt**_ your observations about RemoteIo and one day somebody will have fun dealing with them.